### PR TITLE
Updates to Zipkin 3, Spring Boot 3 and JRE 17

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This updates server modules to Zipkin 3, Spring Boot 3 and JRE 17. It leaves the client side modules at Java 8.

In order to compile the server module, we can no longer compile with JDK 11.

See https://github.com/openzipkin/zipkin/pull/3684

### CODE REVIEW

File: .github/workflows/create_release.yml [hunk 1]
```diff
@@ -24,13 +24,13 @@ jobs:
...
           distribution: 'zulu'  # zulu as it supports a wide version range
-          java-version: '11'  # earliest LTS and last that can compile the 1.6 release profile.
+          java-version: '17'  # earliest LTS and last that can compile the 1.6 release profile.
       - name: Cache local Maven repository
         uses: actions/cache@v3
         with:
           path: ~/.m2/repository
-          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
-          restore-keys: ${{ runner.os }}-maven-
+          key: ${{ runner.os }}-jdk-11-maven-${{ hashFiles('**/pom.xml') }}
```
Comment (reta) [inline .github/workflows/create_release.yml:32]:
```suggestion
          key: ${{ runner.os }}-jdk-17-maven-${{ hashFiles('**/pom.xml') }}
```

File: aws-junit/src/main/java/zipkin2/junit/aws/AmazonSQSExtension.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ public AmazonSQSExtension() {
       client = null;
     }
 
-    if (server == null) {
+    if (server != null) {
```
Comment (reta) [inline aws-junit/src/main/java/zipkin2/junit/aws/AmazonSQSExtension.java:78]: 👍

**Review by reta — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
thanks again @reta!

File: .github/workflows/create_release.yml [hunk 3]
```diff
@@ -24,13 +24,13 @@ jobs:
...
-          java-version: '11'  # earliest LTS and last that can compile the 1.6 release profile.
+          java-version: '17'  # earliest LTS and last that can compile the 1.6 release profile.
       - name: Cache local Maven repository
         uses: actions/cache@v3
         with:
           path: ~/.m2/repository
-          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
-          restore-keys: ${{ runner.os }}-maven-
+          key: ${{ runner.os }}-jdk-17-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-jdk-11-maven-
```
Comment (reta) [inline .github/workflows/create_release.yml:33]: Same, super sorry about that:
```suggestion
          restore-keys: ${{ runner.os }}-jdk-17-maven-
```

**Comment by codefromthecrypt** [comment]
checked docker from master tag is expected. releasing!

**Pull request merged**